frame/staking-rewards/src/test/mod.rs

```diff
@@ -246,6 +265,202 @@ fn test_extend_stake_amount() {
+			(ALICE, StakingRewards::pool_count(), 100_500u32.into(), ONE_HOUR, 100, 200, 50);
+
+		let rewards_pool =
+			StakingRewards::pools(StakingRewards::pool_count()).expect("rewards_pool expected");
+		let staked_asset_id = rewards_pool.asset_id;
+		mint_assets(
+			vec![staker, StakingRewards::pool_account_id(&pool_id)],
+			vec![staked_asset_id]
+				.iter()
+				.chain(rewards_pool.rewards.iter().map(|(asset_id, _inflation)| asset_id))
+				.cloned()
+				.collect::<Vec<_>>(),
+			amount * 2,
+		);
+		update_total_rewards_and_total_shares_in_rewards_pool(pool_id, total_rewards, total_shares);
+
+		assert_ok!(StakingRewards::stake(Origin::signed(staker), pool_id, amount, duration_preset));
+		let stake_id = StakingRewards::stake_count();
+		assert_eq!(balance(staked_asset_id, &staker), amount);
+
+		let mut stake = StakingRewards::stakes(stake_id).expect("stake expected");
+		let unlock_penalty = stake.lock.unlock_penalty;
+		stake.reductions = update_reductions(stake.reductions, claim);
+		Stakes::<Test>::insert(stake_id, stake);
+
+		assert_ok!(StakingRewards::unstake(Origin::signed(staker), stake_id));
+		assert_eq!(StakingRewards::stakes(stake_id), None);
+		assert_last_event::<Test, _>(|e| {
+			matches!(
+				e.event,
+				Event::StakingRewards(crate::Event::Unstaked { owner, position_id })
+					if owner == staker && position_id == stake_id
+			)
+		});
+
+		let penalty = unlock_penalty.mul_ceil(amount);
+		let claim_with_penalty = (Perbill::one() - unlock_penalty).mul_ceil(claim);
+		let rewards_pool =
+			StakingRewards::pools(StakingRewards::pool_count()).expect("rewards_pool expected");
+		assert_eq!(balance(staked_asset_id, &staker), amount * 2 - penalty);
+		assert_eq!(
+			balance(staked_asset_id, &StakingRewards::pool_account_id(&pool_id)),
+			amount * 2 + penalty
+		);
+		for (asset_id, _) in rewards_pool.rewards.iter() {
+			assert_eq!(balance(*asset_id, &staker), amount * 2 + claim_with_penalty);
+			assert_eq!(
+				balance(*asset_id, &StakingRewards::pool_account_id(&pool_id)),
+				amount * 2 - claim_with_penalty
+			);
+		}
+	});
+}
+
+#[test]
+fn unstake_in_case_of_not_zero_claims_and_not_early_unlock_should_work() {
+	new_test_ext().execute_with(|| {
+		System::set_block_number(1);
+		assert_ok!(StakingRewards::create_reward_pool(Origin::root(), get_default_reward_pool()));
+		let (staker, pool_id, amount, duration_preset, total_rewards, total_shares, claim) =
+			(ALICE, StakingRewards::pool_count(), 100_500u32.into(), ONE_HOUR, 100, 200, 50);
+
+		let rewards_pool =
+			StakingRewards::pools(StakingRewards::pool_count()).expect("rewards_pool expected");
+		let staked_asset_id = rewards_pool.asset_id;
+		mint_assets(
+			vec![staker, StakingRewards::pool_account_id(&pool_id)],
+			vec![staked_asset_id]
+				.iter()
+				.chain(rewards_pool.rewards.iter().map(|(asset_id, _inflation)| asset_id))
+				.cloned()
+				.collect::<Vec<_>>(),
+			amount * 2,
```

Comment:
Why not
```suggestion
			amount,
```
?

Reply:
`stake` calls `transfer` with `keep_alive = true`.
So we will get `NotEnoughAssets` error below when will do `stake(amount)`.